vyxal/elements.py

```diff
@@ -859,8 +859,12 @@ def gen():
                 return
 
             diag_num = 0
-            lhs_max = len(lhs) - 1 if isinstance(lhs, list) else None
-            rhs_max = len(rhs) - 1 if isinstance(rhs, list) else None
+            lhs_max = (
+                len(lhs) - 1 if vy_type(lhs, simple=True) == list else None
+            )
+            rhs_max = (
+                len(rhs) - 1 if vy_type(rhs, simple=True) == list else None
+            )
```

Comment:
The `len(lhs) if isinstance(lhs, list)` is so that `len` is only called on lists, not `LazyList`s. This change'll break infinite it for `LazyList`s.